---
title: "Data Visualization"
author: "Pan"
date: "25 August 2017"
output: slidy_presentation 
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE)
```

## Research question

```{r echo=FALSE, out.width='100%'}
knitr::include_graphics('/projects/Goeteborg project/20160510/Gothenburg project/figure for publication/figure1/Figure1.new.jpg')
```


## Information want to present

- Differential expressed gene name
- Time
- Gene position
- DNA methylation 

## R package: circlize

Load Data
```{r, include = FALSE, echo=FALSE}
library(xlsx)
```

```{r , echo = TRUE}
DMR<-read.xlsx("/projects/Goeteborg project/20160510/Gothenburg project/figure for publication/figure5/INTERGRATE_NEW.xlsx",sheetIndex = 1)
gene<-read.xlsx("/projects/Goeteborg project/20160510/Gothenburg project/figure for publication/figure5/INTERGRATE_NEW.xlsx",sheetIndex = 2)
head(gene)
gene$chr<-paste0("chr",gene$chr)
DMR$chr<-paste0("chr",DMR$chr)
gene$week<-factor(gene$week,levels = c("W1","W4","W12/16"))
```

## Mouse genome

```{r, echo=TRUE, message=FALSE, warning=FALSE}
library(circlize)
circos.clear()
circos.par(gap.degree = 2, start.degree = 80)
circos.initializeWithIdeogram(chromosome.index = c(paste0("chr", 1:19),"chrX"), plotType = c("ideogram", "labels"),ideogram.height = 0.03,species = "mm10")
```

## input data

```{r, echo=TRUE, message=FALSE, warning=FALSE}
for(i in seq_len(nrow(gene))) {
  gene[i, "theta1"] = circlize(gene[i, 2], 1, sector.index = gene[i, 1])[1, 1]
  gene[i, "theta2"] = circlize(gene[i, 3], 1, sector.index = gene[i, 1])[1, 1]
}
r = circlize:::get_most_inside_radius()
circos.clear()
```
make different colors for different weeks
```{r, echo=TRUE, message=FALSE, warning=FALSE}
circos.clear()
circos.par(gap.degree = 2, start.degree = 80)
circos.initializeWithIdeogram(chromosome.index = c(paste0("chr", 1:19),"chrX"), plotType = c("ideogram", "labels"),ideogram.height = 0.03,species = "mm10")
par(new = TRUE)
W12_col<-rgb(97,156,255,0.6,max=255)
week_color=c("#F8766D33","#00BA3833","#619CFF33")
names(week_color)=factor(c("W1","W4","W12/16"),levels = c("W1","W4","W12/16"))


circos.par(cell.padding = c(0.02, 0, 0.02, 0), gap.degree = c(rep(1, nrow(gene)-1), 7), 
           start.degree = 80, points.overflow.warning = FALSE)
circos.initialize(factors = factor(row.names(gene), levels = row.names(gene)), 
                  xlim = as.matrix(gene[, 2:3]))

circos.track(ylim = c(0, 1), panel.fun = function(x, y) {
  
  si = get.cell.meta.data("sector.index")
  chr = gsub("^(chr.*?)-.*$", "\\1", si)
  theta1 = gene[si, "theta1"]
  theta2 = gene[si, "theta2"]
  
  xlim = get.cell.meta.data("cell.xlim")
  ylim = get.cell.meta.data("cell.ylim")
  
  cell.top.radius = get.cell.meta.data("cell.top.radius")
  
  # map from polar coordinate system to data coordinate system
  df = reverse.circlize(c(theta1, theta2), c(cell.top.radius, cell.top.radius))
  
  x21 = df[1, 1]
  x22 = df[2, 1]
  y21 = df[1, 2]
  y22 = df[2, 2]
  x11 = xlim[1]
  x12 = xlim[2]
  y11 = ylim[1]
  y12 = ylim[1]
  circos.polygon(c(x11, x11, x21, x21, x22, x22, x12, x12, x11),
                 c(y11, (y21 - y11)/3, (y21 - y11)/3*2, y21, y22, (y22 - y12)/3*2, (y22 - y12)/3, y12, y11), col = week_color[gene[si, "week"]])
  
}, track.margin = c(0, 1 - r), cell.padding = c(0, 0, 0, 0), bg.border = NA, track.height = 0.1)
```

## DNA methylation

```{r, echo=TRUE, message=FALSE, warning=FALSE}
circos.clear()
circos.par(gap.degree = 2, start.degree = 80)
circos.initializeWithIdeogram(chromosome.index = c(paste0("chr", 1:19),"chrX"), plotType = c("ideogram", "labels"),ideogram.height = 0.03,species = "mm10")
par(new = TRUE)
W12_col<-rgb(97,156,255,0.6,max=255)
week_color=c("#F8766D33","#00BA3833","#619CFF33")
names(week_color)=factor(c("W1","W4","W12/16"),levels = c("W1","W4","W12/16"))


circos.par(cell.padding = c(0.02, 0, 0.02, 0), gap.degree = c(rep(1, nrow(gene)-1), 7), 
           start.degree = 80, points.overflow.warning = FALSE)
circos.initialize(factors = factor(row.names(gene), levels = row.names(gene)), 
                  xlim = as.matrix(gene[, 2:3]))

circos.track(ylim = c(0, 1), panel.fun = function(x, y) {
  
  si = get.cell.meta.data("sector.index")
  chr = gsub("^(chr.*?)-.*$", "\\1", si)
  theta1 = gene[si, "theta1"]
  theta2 = gene[si, "theta2"]
  
  xlim = get.cell.meta.data("cell.xlim")
  ylim = get.cell.meta.data("cell.ylim")
  
  cell.top.radius = get.cell.meta.data("cell.top.radius")
  
  # map from polar coordinate system to data coordinate system
  df = reverse.circlize(c(theta1, theta2), c(cell.top.radius, cell.top.radius))
  
  x21 = df[1, 1]
  x22 = df[2, 1]
  y21 = df[1, 2]
  y22 = df[2, 2]
  x11 = xlim[1]
  x12 = xlim[2]
  y11 = ylim[1]
  y12 = ylim[1]
  circos.polygon(c(x11, x11, x21, x21, x22, x22, x12, x12, x11),
                 c(y11, (y21 - y11)/3, (y21 - y11)/3*2, y21, y22, (y22 - y12)/3*2, (y22 - y12)/3, y12, y11), col = week_color[gene[si, "week"]])
  
}, track.margin = c(0, 1 - r), cell.padding = c(0, 0, 0, 0), bg.border = NA, track.height = 0.1)

first_sector = get.all.sector.index()[1]
max_abs = max(abs(c(DMR$diff)))
max_abs = ceiling(max_abs*10)/10
max_abs=0.75
gene_x<-gene_y<-NULL
circos.track(ylim = c(-max_abs, max_abs), panel.fun = function(x, y) {
  si = get.cell.meta.data("sector.index")
  chr = gsub("^(chr\\d+).*$", "\\1", si)
  xlim = get.cell.meta.data("cell.xlim")
  ylim = get.cell.meta.data("cell.ylim")
  
  circos.rect(xlim[1], ylim[1], xlim[2], ylim[2], col = week_color[gene[si, "week"]])
  for(h in seq(-max_abs, max_abs, by = 0.25)) {
    circos.lines(xlim, c(h, h), lty = 3, col = "#AAAAAA")
  }
  
  circos.lines(xlim, c(0, 0), lty = 3, col = "#888888")
  
  subset = DMR[DMR$gene_index == si, , drop = FALSE]
  if(nrow(subset) > 0) {
    circos.points((subset[[2]] + subset[[3]])/2, subset$diff, 
                  col = ifelse(subset$diff > 0, "#E41A1C", "#0B2161"), pch = 16, cex = 0.5)
  }
  
  gene_x<<-append(gene_x,mean(xlim))
  gene_y<<-append(gene_y,mean(ylim))
  
}, bg.border = 1, track.height = 0.15)
circos.yaxis(side = "left", at = seq(-0.5, 0.5, by = 0.5), sector.index = first_sector,
             labels.cex = 0.4)
xlim = get.cell.meta.data("cell.xlim", sector.index = first_sector)
ylim = get.cell.meta.data("cell.ylim", sector.index = first_sector)
circos.text(xlim[1], mean(ylim), "d", facing = "clockwise", niceFacing = TRUE, cex = 0.6, 
            adj = c(0,degree(4.5)), sector.index = first_sector)
r = circlize:::get_most_inside_radius()
```

## Labeling

```{r, echo=FALSE, message=FALSE, warning=FALSE,results=FALSE}
library(xlsx)
DMR<-read.xlsx("/projects/Goeteborg project/20160510/Gothenburg project/figure for publication/figure5/INTERGRATE_NEW.xlsx",sheetIndex = 1)
gene<-read.xlsx("/projects/Goeteborg project/20160510/Gothenburg project/figure for publication/figure5/INTERGRATE_NEW.xlsx",sheetIndex = 2)
head(DMR)
gene$chr<-paste0("chr",gene$chr)
DMR$chr<-paste0("chr",DMR$chr)
gene$week<-factor(gene$week,levels = c("W1","W4","W12/16"))

#plot the base
circos.clear()
circos.par(gap.degree = 2, start.degree = 80)
circos.initializeWithIdeogram(chromosome.index = c(paste0("chr", 1:19),"chrX"), plotType = c("ideogram", "labels"),
                              ideogram.height = 0.03,species = "mm10")

#target
for(i in seq_len(nrow(gene))) {
  gene[i, "theta1"] = circlize(gene[i, 2], 1, sector.index = gene[i, 1])[1, 1]
  gene[i, "theta2"] = circlize(gene[i, 3], 1, sector.index = gene[i, 1])[1, 1]
}
head(gene)
#cell.top.radius = get.cell.meta.data("cell.top.radius")
#df = reverse.circlize(c(gene$theta1, gene$theta2), c(cell.top.radius, cell.top.radius))



r = circlize:::get_most_inside_radius()
circos.clear()
#week color
par(new = TRUE)
W12_col<-rgb(97,156,255,0.6,max=255)
week_color=c("#F8766D33","#00BA3833","#619CFF33")
names(week_color)=factor(c("W1","W4","W12/16"),levels = c("W1","W4","W12/16"))


circos.par(cell.padding = c(0.02, 0, 0.02, 0), gap.degree = c(rep(1, nrow(gene)-1), 7), 
           start.degree = 80, points.overflow.warning = FALSE)
circos.initialize(factors = factor(row.names(gene), levels = row.names(gene)), 
                  xlim = as.matrix(gene[, 2:3]))

circos.track(ylim = c(0, 1), panel.fun = function(x, y) {
  
  si = get.cell.meta.data("sector.index")
  chr = gsub("^(chr.*?)-.*$", "\\1", si)
  theta1 = gene[si, "theta1"]
  theta2 = gene[si, "theta2"]
  
  xlim = get.cell.meta.data("cell.xlim")
  ylim = get.cell.meta.data("cell.ylim")
  
  cell.top.radius = get.cell.meta.data("cell.top.radius")
  
  # map from polar coordinate system to data coordinate system
  df = reverse.circlize(c(theta1, theta2), c(cell.top.radius, cell.top.radius))
  
  x21 = df[1, 1]
  x22 = df[2, 1]
  y21 = df[1, 2]
  y22 = df[2, 2]
  x11 = xlim[1]
  x12 = xlim[2]
  y11 = ylim[1]
  y12 = ylim[1]
  circos.polygon(c(x11, x11, x21, x21, x22, x22, x12, x12, x11),
                 c(y11, (y21 - y11)/3, (y21 - y11)/3*2, y21, y22, (y22 - y12)/3*2, (y22 - y12)/3, y12, y11), 
                 col = week_color[gene[si, "week"]])
  
 
  
}, track.margin = c(0, 1 - r), cell.padding = c(0, 0, 0, 0), bg.border = NA, track.height = 0.1)


#methylation
first_sector = get.all.sector.index()[1]
max_abs = max(abs(c(DMR$diff)))
max_abs = ceiling(max_abs*10)/10
max_abs=0.75
gene_x<-gene_y<-NULL
circos.track(ylim = c(-max_abs, max_abs), panel.fun = function(x, y) {
  si = get.cell.meta.data("sector.index")
  chr = gsub("^(chr\\d+).*$", "\\1", si)
  xlim = get.cell.meta.data("cell.xlim")
  ylim = get.cell.meta.data("cell.ylim")
  
  circos.rect(xlim[1], ylim[1], xlim[2], ylim[2], col = week_color[gene[si, "week"]])
  for(h in seq(-max_abs, max_abs, by = 0.25)) {
    circos.lines(xlim, c(h, h), lty = 3, col = "#AAAAAA")
  }
  
  circos.lines(xlim, c(0, 0), lty = 3, col = "#888888")
  
  subset = DMR[DMR$gene_index == si, , drop = FALSE]
  if(nrow(subset) > 0) {
    circos.points((subset[[2]] + subset[[3]])/2, subset$diff, 
                  col = ifelse(subset$diff > 0, "#E41A1C", "#0B2161"), pch = 16, cex = 0.5)
  }
  
  gene_x<<-append(gene_x,mean(xlim))
  gene_y<<-append(gene_y,mean(ylim))
  
}, bg.border = 1, track.height = 0.15)
circos.yaxis(side = "left", at = seq(-0.5, 0.5, by = 0.5), sector.index = first_sector,
             labels.cex = 0.4)
xlim = get.cell.meta.data("cell.xlim", sector.index = first_sector)
ylim = get.cell.meta.data("cell.ylim", sector.index = first_sector)
circos.text(xlim[1], mean(ylim), "d", facing = "clockwise", niceFacing = TRUE, cex = 0.6, 
            adj = c(0,degree(4.5)), sector.index = first_sector)
r = circlize:::get_most_inside_radius()


#gene


circos.track(ylim = c(-0, 1), panel.fun = function(x, y) {
  si = get.cell.meta.data("sector.index")
  chr = gsub("^(chr\\d+).*$", "\\1", si)
  xlim = get.cell.meta.data("cell.xlim")
  ylim = get.cell.meta.data("cell.ylim")
  circos.text(mean(xlim), mean(ylim), labels=gene[si,"gene_name"],adj = c(0,0.5), 
              facing = "reverse.clockwise",cex=0.5,niceFacing = TRUE)
  
}, bg.border = NA, track.height = 0.05)
```

## Modified version

```{r echo=FALSE, out.width='100%'}
knitr::include_graphics('/projects/job/CCR/CCR/Modified.png')
```

## Other figure
```{r echo=FALSE, out.width='100%'}
knitr::include_graphics('/projects/Twin data project/manuscript/Figures/figure2/figure2.png')
```

## Other figure
```{r echo=FALSE, out.width='100%'}
knitr::include_graphics('/projects/Twin data project/manuscript/Figures/figure4/figure4.png')
```

## Other figure
```{r echo=FALSE, out.width='100%'}
knitr::include_graphics('/projects/job/CCR/CCR/Phylum_all_bar_0410.png')
```